metrics/metrics.go

```diff
@@ -28,7 +77,28 @@ func NewGarmCollector(r *runner.Runner) *GarmCollector {
 			"Health of the runner",
 			[]string{"hostname", "controller_id"}, nil,
 		),
+		cachedControllerInfo: controllerInfo,
+	}, nil
+}
+
+type GarmCollector struct {
+	healthMetric         *prometheus.Desc
+	instanceMetric       *prometheus.Desc
+	runner               *runner.Runner
+	cachedControllerInfo params.ControllerInfo
+	mux                  sync.Mutex
+}
+
+func (c *GarmCollector) controllerInfo() params.ControllerInfo {
+	controllerInfo, err := c.runner.GetControllerInfo(auth.GetAdminContext())
+	if err != nil {
+		log.Printf("error getting controller info: %s", err)
+		return c.cachedControllerInfo
 	}
+	c.mux.Lock()
+	defer c.mux.Unlock()
+	c.cachedControllerInfo = controllerInfo
```

Comment:
Why is this cached here again? This will only help if runner.GetControllerInfo fails but did work some rounds before, right ?

Reply:
The ```os.Hostname()``` call can fail, which means we'll end up with no hostname when it does. At least for that call. The cache is to fall back on previous value. What I dislike is that I had to duplicate this in 2 places. I Think I'll add a retry strategy instead and remove the cache.

Reply:
I'd still like to know when something fails and react accordingly, than to ignore the error.

Reply:
Removed the caches and just added a retry in ```GetControllerInfo()```. If it errs there, we just surface that error further along and log it.